platform/servers_list.cpp

```diff
@@ -10,33 +10,58 @@
 
 namespace downloader
 {
-// Returns false if can't parse urls. Note that it also clears outUrls.
-bool ParseServerList(std::string const & jsonStr, std::vector<std::string> & outUrls)
+std::optional<MetaConfig> ParseMetaConfig(std::string const & jsonStr)
 {
-  outUrls.clear();
+  MetaConfig outMetaConfig;
   try
   {
-    base::Json root(jsonStr.c_str());
-    for (size_t i = 0; i < json_array_size(root.get()); ++i)
+    const base::Json root(jsonStr.c_str());
+    const json_t * servers;
+    if (json_is_object(root.get()))
     {
-      char const * url = json_string_value(json_array_get(root.get(), i));
+      // New format:
+      // {
+      //   "servers": ["http://url1", "http://url2", "http://url3"],
+      //   "settings": {
+      //      "key1": "value1",
+      //      "key2": "http://url"
+      //    }
+      // }
+
+      json_t * settings = json_object_get(root.get(), "settings");
+      const char * key;
+      const json_t * value;
+      json_object_foreach(settings, key, value)
+      {
+        const char * valueStr = json_string_value(value);
+        if (key && value)
+          outMetaConfig.m_settings[key] = valueStr;
+      }
+
+      servers = json_object_get(root.get(), "servers");
+    }
+    else
+    {
+      // Old format:
+      // ["http://url1", "http://url2", "http://url3"]
+      servers = root.get();
+    }
+
+    for (size_t i = 0; i < json_array_size(servers); ++i)
+    {
+      char const * url = json_string_value(json_array_get(servers, i));
       if (url)
-        outUrls.push_back(url);
+        outMetaConfig.m_serversList.push_back(url);
     }
```

Comment:
1. return from here: `return { serversList, {} }`
2. No need to declare metaconfig above.

Reply:
This code is shared between two cases. Proposed refactoring will almost double the size of the function.